src/openms/source/CHEMISTRY/SvmTheoreticalSpectrumGeneratorSet.cpp

```diff
@@ -86,54 +59,56 @@ namespace OpenMS
       filename = File::find(filename);
     }
 
+    TextFile file(filename);
+    // Boost::empty()
+    if (std::begin(file) == std::end(file)) { return; }// no data to load
+
+    // skip header line
+    const auto begin_from_skipped_header = std::next(std::begin(file));
+
     Param sim_param = SvmTheoreticalSpectrumGenerator().getDefaults();
 
-    TextFile file(filename);
-    TextFile::ConstIterator it = file.begin();
+    const auto process_content = [this, filename, &sim_param] (TextFile::Iterator::value_type i)
```

Comment:
@timosachsenberg I am not sure I agree with this, this is turning a simple loop into a function without any real benefit. As a simple "code maintenance" I dont really agree with such a change

Reply:
agree

Reply:
The `lambda` form helps encapsulate the `for-loop` form. It reveals the `for-loop` form's dependencies to help with further refactoring. 
The basic `std::for_each` change was to help keep the PR small. Further refactoring can be done however this PR is to check if refactoring would be accepted or not.